homeassistant/components/shelly/cover.py

```diff
@@ -199,19 +202,42 @@ def is_opening(self) -> bool:
         """Return if the cover is opening."""
         return cast(bool, self.status["state"] == "opening")
 
+    def launch_update_task(self) -> None:
+        """Launch the update position task if needed."""
+        if not self._update_task:
+            self._update_task = (
+                self.coordinator.config_entry.async_create_background_task(
+                    self.hass, self.update_position(), "Shelly cover update"
+                )
+            )
+
+    async def update_position(self) -> None:
+        """Update the cover position every second."""
+        await self.coordinator.device.update_status()
```

Comment:
We should use try...except when calling `update_status()` and set
```python
self._update_task = None
```
when exception occurs.

Reply:
After some testing with exceptions, I ended up using try: finally: but also checking if the task is None OR done() before launching another one. As the exception is happening in a task, setting _update_task to None is not reliably working with the main thread but the task is considered done in that case

Reply:
Quick question, should I log any exception occurring at this point ? For now I have no except:

Reply:
@bieniu Hi, I would like your input on the question above and I think the PR is ready for review again. Thanks !